docs/xks/operator-guide/blue-green.md

```diff
@@ -0,0 +1,59 @@
+---
+id: blue green
+title: Cluster blue green
+---
+
+For different reasons you might want to create a completely new cluster, this can be due to many reasons like:
+
+- Broken cluster
+- Test patch the cluster
+- Major breaking change
+
+Thanks to this we support performing blue green on a entire kubernetes cluster, this is applicable for both Azure and AWS.
+These docs is intended for both clouds, the main difference is in the naming.
+
+Today we do not support any way of only doing blue green on a specific environment.
+If you need to perform blue green on QA you should do it in dev and prod as well.
+We think that the risk is to great that you by mistake would get drift between the modules used in the different clusters would be to great.
+
+## Workflow
+
+> We assume that the workloads on the clusters is stateless and can run multiple instances.
+
+- Setup a new cluster in the target environment using terraform
+- Verify that the new cluster is functioning as intended
+  - You **won't** be able to verify any ingress
+  - You **won't** be able to use AZAD-proxy in the newly created cluster
+- Change the txt files over to the newly created cluster
+- Verify that the ingress traffic is migrated to the new cluster and it's working as intended
+- Destroy the old cluster using terraform
+
+## DNS migration
+
+We have written a small script to make the migration easier for us.
+As always use at your own risk and make sure that you understand what the script does.
+
+Our recommendation is that you migrate one DNS record manually and verify that the ingress and the new cluster is working as intended,
+when you know that you can run the script.
+
+### Azure
+
+```bash
+ENVIRONMENT="dev"
+OLD_OWNER_ID="${ENVIRONMENT}-aks1"
+NEW_OWNER_ID="${ENVIRONMENT}-aks2"
+RESOURCE_GROUP_NAME="rg-${ENVIRONMENT}-we-aks"
+ZONE_NAME="${ENVIRONMENT}.domain.se"
+ZONE_RECORDS=$(az network dns record-set txt list -g ${RESOURCE_GROUP_NAME} -z ${ZONE_NAME} | jq -rc '.[]')
+ZONE_RECORDS_CSV_ARRAY=( $(jq -rc '. | [.name, (.txtRecords[0].value[0] | @base64)] | join(";")' <<< "${ZONE_RECORDS}") )
+for ZONE_RECORD_CSV in "${ZONE_RECORDS_CSV_ARRAY[@]}"; do
+  ZONE_RECORD_NAME=$(awk -F';' '{print $1}' <<< $ZONE_RECORD_CSV)
+  OLD_ZONE_TXT_VALUE=$(awk -F';' '{print $2}' <<< $ZONE_RECORD_CSV | base64 -d)
```

Comment:
I just copied a script from Simon, I don't have time to make the script better right now.
If you have the time I would love a update of the PR, else I think it will have to do for now.